src/ctap2/mod.rs

```diff
@@ -809,5 +813,497 @@ pub(crate) fn set_or_change_pin_helper(
+                            );
+                            continue;
+                        }
+                    }
+                    BioEnrollmentCommand::EnumerateEnrollments => {
+                        let list = result.template_infos.iter().map(|x| x.into()).collect();
+                        callback.call(Ok(ManageResult::BioEnrollment(
+                            BioEnrollmentResult::EnrollmentList(list),
+                        )));
+                        return true;
+                    }
+                    BioEnrollmentCommand::SetFriendlyName(_)
+                    | BioEnrollmentCommand::RemoveEnrollment(_)
+                    | BioEnrollmentCommand::CancelCurrentEnrollment => {
+                        callback.call(Ok(ManageResult::Success));
+                        return true;
+                    }
+                    BioEnrollmentCommand::GetFingerprintSensorInfo => {
+                        let fingerprint_kind = unwrap_option!(result.fingerprint_kind, callback);
+                        let max_capture_samples_required_for_enroll = unwrap_option!(
+                            result.max_capture_samples_required_for_enroll,
+                            callback
+                        );
+                        let max_template_friendly_name =
+                            unwrap_option!(result.max_template_friendly_name, callback);
+                        callback.call(Ok(ManageResult::BioEnrollment(
+                            BioEnrollmentResult::FingerprintSensorInfo(FingerprintSensorInfo {
+                                fingerprint_kind,
+                                max_capture_samples_required_for_enroll,
+                                max_template_friendly_name,
+                            }),
+                        )));
+                        return true;
+                    }
+                };
+            }
+            Err(e) => {
+                handle_errors!(e, status, callback, pin_uv_auth_result, skip_uv, skip_puap);
+            }
+        }
+    }
+    false
+}
+
+pub(crate) fn credential_management(
+    dev: &mut Device,
+    command: CredManagementCmd,
+    status: Sender<crate::StatusUpdate>,
+    callback: StateCallback<crate::Result<crate::ManageResult>>,
+    alive: &dyn Fn() -> bool,
+) -> bool {
+    let mut skip_uv = false;
+    let authinfo = match dev.get_authenticator_info() {
+        Some(i) => i.clone(),
+        None => {
+            callback.call(Err(HIDError::DeviceNotInitialized.into()));
+            return false;
+        }
+    };
+
+    if authinfo.options.cred_mgmt != Some(true)
+        && authinfo.options.credential_mgmt_preview != Some(true)
+    {
+        callback.call(Err(AuthenticatorError::HIDError(
+            HIDError::UnsupportedCommand,
+        )));
+        return false;
+    }
+
+    let use_legacy_preview = authinfo.options.cred_mgmt != Some(true);
+
+    let mut cred_management = match command {
```

Comment:
I guess we need to pick something due to how `determine_puap_if_needed` is implemented. It's safe to assume that the first command is `GetCredentials`.